packages/shared/src/components/buttons/QuaternaryButton.tsx

```diff
@@ -1,52 +1,48 @@
 import React, {
-  HTMLAttributes,
-  LegacyRef,
+  forwardRef,
   ReactElement,
+  Ref,
+  useImperativeHandle,
   useRef,
   useState,
 } from 'react';
 import classNames from 'classnames';
-import { Button, ButtonProps } from './Button';
+import { Button, AllowedTags, ButtonProps } from './Button';
 
 type QuandaryButtonProps = {
   id: string;
   reverse?: boolean;
   responsiveLabelClass?: string;
 };
 
-export function QuaternaryButton<Tag extends keyof JSX.IntrinsicElements>({
-  id,
-  children,
-  style,
-  className,
-  reverse,
-  responsiveLabelClass,
-  tag,
-  ...props
-}: ButtonProps<Tag> & QuandaryButtonProps): ReactElement {
-  let labelProps: HTMLAttributes<HTMLLabelElement> = {};
-  let buttonProps: {
-    className?: string;
-    innerRef?: LegacyRef<HTMLButtonElement>;
-  } = {};
-  if (tag === 'a') {
-    const buttonRef = useRef<HTMLButtonElement>(null);
-    const [isHovered, setIsHovered] = useState(false);
-    const onLabelClick = (event: React.MouseEvent<HTMLLabelElement>): void => {
-      event.preventDefault();
-      buttonRef.current.click();
-    };
-    labelProps = {
-      onMouseOver: () => setIsHovered(true),
-      onMouseLeave: () => setIsHovered(false),
-      onClick: onLabelClick,
-    };
-    buttonProps = {
-      className: isHovered && 'hover',
-      innerRef: buttonRef,
-    };
-  }
-
+export default function QuaternaryButtonComponent<TagName extends AllowedTags>(
```

Comment:
Why do we export both components? I don't think we need the non forwardRef version.
This is very confusing